nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -488,38 +504,38 @@ in {
         nextcloud-cron = {
           environment.NEXTCLOUD_CONFIG_DIR = "${cfg.home}/config";
           serviceConfig.Type = "oneshot";
-          serviceConfig.User = "nextcloud";
+          serviceConfig.User = cfg.user;
           serviceConfig.ExecStart = "${phpPackage}/bin/php -f ${cfg.package}/cron.php";
         };
         nextcloud-update-plugins = mkIf cfg.autoUpdateApps.enable {
           serviceConfig.Type = "oneshot";
           serviceConfig.ExecStart = "${occ}/bin/nextcloud-occ app:update --all";
-          serviceConfig.User = "nextcloud";
+          serviceConfig.User = cfg.user;
           startAt = cfg.autoUpdateApps.startAt;
         };
       };
 
       services.phpfpm = {
         pools.nextcloud = {
-          user = "nextcloud";
-          group = "nginx";
+          user = cfg.user;
+          inherit group;
           phpOptions = phpOptionsStr;
           phpPackage = phpPackage;
           phpEnv = {
             NEXTCLOUD_CONFIG_DIR = "${cfg.home}/config";
             PATH = "/run/wrappers/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin";
           };
           settings = mapAttrs (name: mkDefault) {
-            "listen.owner" = "nginx";
-            "listen.group" = "nginx";
+            "listen.owner" = cfg.user;
```

Comment:
This isn't what we want. The `user` option should specify the user that the `nextcloud` service runs as, not the user the web server runs as. `"listen.owner"` is intended to be the user the web server is run as.

Reply:
Then I should change the user back to `nextcloud` (due to the other discussions).
And the `listen.owner/group` should be set to `config.services.nginx.user/group` if nginx is enabled.
But what if nginx is disabled? Should we add optional options to configure the owner/group for the webservice?